Fix overlapping gallery slider in Firefox

Saira-A opened the pull request:
This fixes this issue on Firefox
![image](https://github.com/user-attachments/assets/245c97b8-5e6c-4539-9540-10c544164aa7)

by increasing the width of the entire bar: 
![image](https://github.com/user-attachments/assets/6bf31b61-2576-479f-9739-4df4f317299d)

Ideally it needs to be fixed properly so that the bar can remain at the regular width without overlap

demiankatz commented on src/content-handlers/iiif/modules/uv-contentleftpanel-module/css/styles.less:
```diff
@@ -156,3 +156,9 @@
     }
   }
 }
+@-moz-document url-prefix() {
+  .iiif-gallery-component .header input[type="range"] {
+    margin-top: 7px;
+    width: 300px;
+  }
+}
```
Does this make any difference?

```
.iiif-gallery-component .header input[type=range]::-moz-range-track {
  width: 100px;
  z-index: -1;
}
```

(I haven't tested, just wondering if it might help; I took inspiration from [this StackOverflow](https://stackoverflow.com/questions/30723341/different-z-index-for-thumb-and-track-in-html5-range-slider-in-firefox)...)

Saira-A replied:
It does! Thanks @demiankatz. The margin-top property now doesn't work though so I'll try and fix that to move the bar up a bit more 
<img width="192" alt="Screenshot 2024-10-29 at 11 27 54" src="https://github.com/user-attachments/assets/4f084259-15fd-4a6b-835b-b7c174dd9ebf">

demiankatz replied:
Excellent, thanks! Since the width part works, it looks like you might be able to drop the z-index line to simplify further.

Saira-A replied:
Done that, and since margin-top wasn't working with the bar I added it to the buttons instead to move them down so it looks better

demiankatz replied:
Looks reasonable to me, but I'm not seeing the focus highlighting here; I suspect this branch may have been created before the other PR was merged. Do you mind merging the upstream dev branch into this so we can look at all the fixes together in one place?

demiankatz reviewed (changes requested):
Thanks, @Saira-A, this is looking great now!

One question, though -- why did you put this in uv-contentleftpanel-module? Would it be better to put this in the shared module with the other gallery component styles, or did you have to use this location for some reason?

crhallberg commented:
[This line is the issue that's causing the track to run over](https://github.com/UniversalViewer/universalviewer/blob/main/src/content-handlers/iiif/modules/uv-shared-module/css/range.less#L10). We would also improve the styling by making the background transparent. You [read this thorough article](https://www.smashingmagazine.com/2021/12/create-custom-range-input-consistent-browsers/) for more.

If we're going to edit range.less, I'd also prefer to have the Webkit and Moz styles in the same block - makes it much clearer and harder to deviate.

demiankatz commented:
Thanks, @crhallberg, that's very helpful! @Saira-A, do you think you can simplify this based on Chris' observation?

crhallberg commented:
This is how I would simplify the range.less file. There are still some fighting styles in other LESS that will need to be changed for better focus styles and margins but here's my reset.

```LESS
.uv {
	@track-color: @gray-light;
	@thumb-color: @gray;
	@thumb-border: @text-color;

    input[type="range"] {
		-webkit-appearance: none;
		appearance: none;
		background-color: transparent;
		cursor: pointer;
	}

	input[type="range"]::-webkit-slider-runnable-track,
	input[type="range"]::-moz-range-track {
		height: 0.25rem;
		background-color: @track-color;
		border: none;
        border-radius: 999px;
	}

	input[type="range"]::-webkit-slider-thumb,
	input[type="range"]::-moz-range-thumb {
		-webkit-appearance: none;
        appearance: none;
		height: 1rem;
		width: 1rem;
		margin-top: -0.5rem;
		background-color: @thumb-color;
		border: 2px solid @thumb-border;
		border-radius: 999px;
	}
}
```

Saira-A commented:
Thanks @demiankatz @crhallberg!
I’ve made the changes in the ranges.less file, but when I put the css for webkit and moz into the same block it looks fine on Firefox but this happens on Chrome: 
<img width="195" alt="Screenshot 2024-10-30 at 10 44 56" src="https://github.com/user-attachments/assets/31850064-2afc-43df-8bbc-79c421d52b88">

Not sure why since they share the same properties but separating them again fixed it.
I also moved the focus highlight for the slider into the range.less file and for the buttons into buttons.less, and added the margin-top property to the Mozilla buttons otherwise they’re not in line with the slider.

demiankatz reviewed (approved):
Thanks, @Saira-A, this looks good to me!

I'll wait to merge in case @LanieOkorodudu wants to test and/or @crhallberg has any further suggestions.

LanieOkorodudu commented:
@Saira-A @demiankatz, I've tested the fix in Firefox, and it looks good to me as well. Thank you, everyone.

crhallberg reviewed (approved):
I agree! Looks good!
